server/game/cards/12-SoW/IronFoundationsStance.js

```diff
@@ -0,0 +1,45 @@
+import { CardTypes, Players } from '../../Constants.js';
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class IronFoundationsStance extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Lower military skill',
+            condition: () => this.game.isDuringConflict(),
+            target: {
+                cardType: CardTypes.Character,
+                controller: Players.Self,
+                cardCondition: card => card.isParticipating() && card.hasTrait('monk'),
+                gameAction: [
+                    AbilityDsl.actions.cardLastingEffect({
+                        effect: AbilityDsl.effects.cardCannot({
+                            cannot: 'sendHome',
+                            restricts: 'opponentsCardEffects'
+                        })
+                    }),
+                    AbilityDsl.actions.cardLastingEffect({
+                        effect: AbilityDsl.effects.cardCannot({
+                            cannot: 'bow',
+                            restricts: 'opponentsCardEffects'
+                        })
+                    })
+                ]
+            },
+            then: context => {
+                if(this.game.currentConflict.getNumberOfCardsPlayed(context.player, card => card.hasTrait('kiho')) < 2) {
+                    return;
+                }
+                return {
+                    gameAction: AbilityDsl.actions.draw(),
+                    message: '{0} draws a card'
+                };
+            },
+            effect: 'prevent opponents\' actions from bowing or moving home {0}'
+        });
+    }
+}
+
```

Comment:
Doesn't the then make it dependent on the previous actions? Both are lasting effects so they probably always happen, but shouldn't this be a AbilityDsl.actions.conditional action grouped together with the other 2?